File: Client/src/services/EnterpriseService.ts
import type { EnterpriseVM } from '@/interfaces'
import Axios from '@/axios'

export default {
  async $GetEnterprises() {
    return await Axios.get('api/Enterprise/GetEnterprises')
  },
  async $SaveEnterprise(enterprise: EnterpriseVM) {
    return await Axios.post('api/Enterprise/SaveEnterprise', enterprise)
  },
}

File: BussinessLogic/Services/EnterpriseService.cs
using BussinessLogic.Interfaces;
using BussinessLogic.Mappers;
using BussinessLogic.ViewModels;
using DomainLayer.Interfaces;
using DomainLayer.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BussinessLogic.Services
{
    public class EnterpriseService : IEnterpriseService
    {
        private readonly IEnterpriseRepository _enterpriseRepository;
        public EnterpriseService(IEnterpriseRepository enterpriseRepository)
        {
            _enterpriseRepository = enterpriseRepository;
        }
        public async Task<List<EnterpriseVM>> GetEnterprises()
        {
            List<Enterprise> enterprisesEntities = await _enterpriseRepository.GetEnterprises();
            return enterprisesEntities.ToViewList();
        }

        public async Task<bool> SaveEnterprise(EnterpriseVM enterprise)
        {
            Enterprise entity = enterprise.ToEntity();
            return await _enterpriseRepository.SaveEnterprise(entity);
        }
    }
}
